### PR TITLE
tools: rewrite benchmark tool

### PR DESCRIPTION
**Pull request by xiang90**
rewrite the benchmark tool for v3 using cobra.

### CODE REVIEW

File: tools/benchmark/cmd/put.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+	for i := range clients {
+		wg.Add(1)
+		go doPut(clients[i], requests)
+	}
+
+	start := time.Now()
+	for i := 0; i < putTotal; i++ {
+		r := &etcdserverpb.PutRequest{
+			Key:   k,
```
Comment (gyuho) [inline tools/benchmark/cmd/put.go]: @xiang90  Is it intentional to run `put`s with the same key, value?

Comment (xiang90) [inline tools/benchmark/cmd/put.go]: for now, yes. I want to add a optional-flag to specify the number of keys to put.

Comment (gyuho) [inline tools/benchmark/cmd/put.go]: Got it. Thanks. Looks good!

**Comment by xiang90** [comment]
@gyuho Maybe you can also give this tool a try? set up a etcd3 cluster try this tool out?

**Comment by gyuho** [comment]
@xiang90 Sure, I will give it a try.

**Comment by gyuho** [comment]
Tried this locally, and confirmed that it works well. Thanks!

```
./benchmark --clients=1000 --endpoint '127.0.0.1:12378' range foo bar
10000 / 10000 Boooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo! 100.00 % 3s

Summary:
  Total:	3.6638 secs.
  Slowest:	0.4704 secs.
  Fastest:	0.1985 secs.
  Average:	0.3579 secs.
  Requests/sec:	2729.4435

Response time histogram:
  0.199 [1]	|
  0.226 [13]	|
  0.253 [441]	|∎∎∎∎∎∎∎
  0.280 [452]	|∎∎∎∎∎∎∎
  0.307 [1450]	|∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎
  0.334 [757]	|∎∎∎∎∎∎∎∎∎∎∎∎∎
  0.362 [2322]	|∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎
  0.389 [1165]	|∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎
  0.416 [1282]	|∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎
  0.443 [2002]	|∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎
  0.470 [115]	|∎

Latency distribution:
  10% in 0.2832 secs.
  25% in 0.3116 secs.
  50% in 0.3552 secs.
  75% in 0.4127 secs.
  90% in 0.4261 secs.
  95% in 0.4307 secs.
  99% in 0.4679 secs.

```

**Comment by xiang90** [comment]
I am going to merge this in and add watcher bench. This is not perfect for sure (we might want to add more benchmark options for example), but I think this is a good start.

**Pull request merged**